### PR TITLE
Add markdown support for content pages

### PR DESCRIPTION
**Pull request by xmorave2**
As markdown is de facto standard for flat file CMSs nowadays

Todo:

- [ ] Update wiki page when merged

### CODE REVIEW

File: module/VuFind/config/module.config.php [hunk 1]
```diff
@@ -333,6 +333,7 @@
     'service_manager' => [
         'allow_override' => true,
         'factories' => [
+            'MarkdownConverter' => 'VuFind\Service\MarkdownFactory',
```
Comment (demiankatz) [inline module/VuFind/config/module.config.php]: Service names should correspond to class or interface names, so I think this should actually be `League\CommonMark\GithubFlavoredMarkdownConverter` (or maybe better yet, `\League\CommonMark\ConverterInterface`). If you think it's important to have a short service name, you can make `MarkdownConverter` an alias, though I think that's probably just extra unnecessary complexity.

File: module/VuFind/src/VuFind/Controller/ContentController.php [hunk 2]
```diff
@@ -55,28 +55,45 @@ public function contentAction()
             ->getLocale();
         $defaultLanguage = $this->getConfig()->Site->language;
 
+        $types = [
```
Comment (demiankatz) [inline module/VuFind/src/VuFind/Controller/ContentController.php]: Would it make sense to move this to a class property so it's easier to extend with additional types?

File: module/VuFind/src/VuFind/Controller/ContentController.php [hunk 3]
```diff
@@ -55,28 +55,45 @@ public function contentAction()
...
+            }
         }
 
-        if (empty($page) || 'content' === $page
-            || null === $themeInfo->findContainingTheme(
-                "templates/content/$page.phtml"
-            )
-        ) {
-            return $this->notFoundAction($this->getResponse());
+        if ($renderer === 'phtml') {
```
Comment (demiankatz) [inline module/VuFind/src/VuFind/Controller/ContentController.php]: To add to flexibility, as a complement to the suggested "class property listing legal types" idea, what if you create a separate method to build the view for each type, and then load it via convention -- e.g.:

<pre>
$method = 'getViewFor' . ucwords($renderer);
if (is_callable([$this, $method])) {
    return $this->$method($page, $path);
}
</pre>

(Not sure if there's a better way to pass parameters more consistently across different methods...)

File: module/VuFind/config/module.config.php (hunk 1 above)
Comment (xmorave2) [reply module/VuFind/config/module.config.php]: I'll go with `League\CommonMark\ConverterInterface`, thanks. I also don't think we need a short name here.

File: module/VuFind/src/VuFind/Controller/ContentController.php (hunk 3 above)
Comment (xmorave2) [reply module/VuFind/src/VuFind/Controller/ContentController.php]: This does make sense, I'll do it that way.

**Comment by demiankatz** [comment]
Thanks, @xmorave2, this looks great. I've made a couple of minor adjustments, and I've deleted the example vufind.md since I don't think we want to ship that in the default package, but I think this is ready to merge, so I'm going to go ahead and do so.

**Comment by xmorave2** [comment]
thanks @demiankatz!

**Pull request merged**